app/models/foreman_openscap/policy.rb

```diff
@@ -21,14 +21,10 @@ class Policy < ActiveRecord::Base
     validates :name, :presence => true, :uniqueness => true
     validate :ensure_needed_puppetclasses
     validates :period, :inclusion => {:in => %w(weekly monthly custom)},
-              :if                 => Proc.new { |policy| policy.new_record? ? policy.step_index > 3 : !policy.id.blank? }
-    validates :weekday, :inclusion => {:in => Date::DAYNAMES.map(&:downcase)},
-              :if                  => Proc.new { |policy| policy.period == 'weekly' && (policy.new_record? ? policy.step_index > 3 : !policy.id.blank?) }
-    validates :day_of_month, :numericality => {:greater_than => 0, :less_than => 32},
-              :if                          => Proc.new { |policy| policy.period == 'monthly'&& (policy.new_record? ? policy.step_index > 3 : !policy.id.blank?) }
-    validate :valid_cron_line
-    validate :ensure_period_specification_present
+              :if => Proc.new { |policy| policy.should_validate?(3) }
```

Comment:
could you please replace all step indexes in this PR with something like `steps.index('Schedule')`